### DISCUSSION TITLE
Need some help with Modbus Configuration

### DISCUSSION THREAD

**Discussion by paulos9** [artisan-roaster-scope/artisan#1273] [Q&A]
I'm have an Autonics TX4S in my roaster connected vía USB to my Mac and want achieve to get here.![image](https://github.com/artisan-roaster-scope/artisan/assets/146485159/df1af068-a44b-4adf-a7d1-be2aa55d0b0e)

How do I configure the ports, the register number, function, decode, etc. According to my reader. I just need the basic readings for my roaster.

**Comment by bigl00z3** [comment]
Your slave number is the Modbus number for the hardware that you are trying to reach on the Modbus line ( [https://techcult.com/a-complete-guide-to-modbus-slave-principle/](https://techcult.com/a-complete-guide-to-modbus-slave-principle/)
Then you need to know for your particular hardware which register is set for each of the required parameter you are trying to read.
The other fields are also important but may be different for each piece of hardware. Without each parameter correct, you won't be able to read the data (I know its a bit complicated). 

For all of the above, as your PID is not in the supported hardware, you will need to read the [user manual for your PID](https://autonics.se/wp-content/uploads/2018/03/tx4s_en_manual_drw170805aa_171115_hw_171127.pdf) or contact the supplier) for questions. 

You can also inspire yourself from other hardware settings. As an example you can check the posts on my blog ([here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-fuji-pxf-and-test-of-maxwell-mta) and [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf)) for the Fuji PXF and MaxWell MTA PIDs.

If you only nee the basic functions, check under which register your temperature value is (SV) and under which register you should put your set value or process value (PV).

Nico

**Reply by bigl00z3** [reply]
Additional comment. 

Looking again on the Autonics TX4S web site you can download [the communication manual ](https://www.autonics.com/service/data/view/119/241893) for your PID. 
You will need to create a user account to get the pdf (which I didn't), but this should help you finding the registers for the parameters you are looking for.

Note: depending on the user manual, you way have to convert the register's address from hexadecimal to decimal (at least that was the case for the Fuji PXF), if you don't know how, please refer to my blog posts listed above.

Nico

**Comment by MAKOMO** [comment]
Roasters with TX4S often use 9600 baud, 8 bits, odd parity (O) with 1 stopbit or no parity (N) with 2 stopbits. Often the machines come with two PIDs, one configured as slave 1 and the other one as slave 2 holding there process value (PV) in register 1000 accessible via MODBUS function 4 in either C or F. Use Artisans channel 1 and 2 to assign those to ET and BT. Some deliver the data as is as uInt16, others as uInt16 with a factor 10 thus on the Artisan one needs to configure the dividier as 1/10.

**Reply by MAKOMO** [reply]
Note that Artisan offers also a simple MODBUS scanner (SCAN button in the MODBUS tab) to "search" for data and find the correct register numbers to use. However, you need to have the basic (serial) connection parameter and the serial port right to have this scanner talk to the device.

**Reply by paulos9** [reply]
Thank you! I've just did it with the Information both of you delivered and I'm starting my first roast! Thank you so much!
